back/node_watcher/src/tasks/types.ts

```diff
@@ -30,7 +30,12 @@ import {
 
 export interface Task<T> {
   name: string;
-  read(blockHash: Hash, api: ApiPromise): Promise<T>;
+  read(
+    blockHash: Hash,
+    events: EventRecord[],
+    sessionIndex: SessionIndex,
+    api: ApiPromise
+  ): T | Promise<T>;
```

Comment:
In `createOfflineValidator`, I'd actually rather have at the end `return Promise.resolve(result)`, so we keep a consistent `Promise<T>` interface